SyncConnectionStateErrorReset/SyncConnectionStateErrorReset/Managers/AppManager.swift

```diff
@@ -0,0 +1,88 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import RealmSwift
+
+let appId = "<ATLAS_APP_ID>"
+let realmApp = RealmSwift.App(id: appId)
+
+class AppErrorManager: ObservableObject {
+    var error: String?
+    @Published var showingError = false
+
+    // This will initialise the error handler, which will trigger
+    // an alert at any moment of the App cycle.
+    func initErrorHandler() {
+        realmApp.syncManager.errorHandler = { (error, _) in
+            self.error = error.localizedDescription
+            self.showingError = true
```

Comment:
One thing I think we're missing here is showing how to handle specific errors that developers encounter. Here, we're just showing the error. In the Node.js example that the curated example brief references, we're checking for specific error codes: https://github.com/elle-j/realm-nodejs-example/blob/8253de42a60df3bc4966b3ede04d24ffe89af120/src/realm-auth.js#L100

There, we're still just showing the error, but it gives developers some example to work from for structuring error handling for specific types of errors. I wonder if we should show a specific example, like handling an Access Token Expired error, which the Node.js example says is an error code 202?